integrationtest/backend/lucene/src/test/java/org/hibernate/search/integrationtest/backend/lucene/testsupport/util/LuceneTckBackendFeatures.java

```diff
@@ -39,4 +39,29 @@ public boolean projectionPreservesEmptySingleValuedObject(ObjectStructure struct
 	public boolean reliesOnNestedDocumentsForMultiValuedObjectProjection() {
 		return true;
 	}
+
+	public boolean supportsHighlighterUnifiedTypeNoMatchSize() {
+		// Lucene default unified highlighter does not support no-match-size setting.
+		// While in ES a custom highlighter is used that allows for such option.
+		return false;
+	}
+
+	public boolean supportsHighlighterUnifiedTypeMultipleFragmentsAsSeparateItems() {
+		// Lucene default unified highlighter will combine all fragments into one string separating them with `ellipses` parameter.
```

Comment:
Right... since Elasticsearch does things their way for the unified highlighter, we have this for Lucene impl:
```java
public PassageFormatterWithEncoder(String preTag, String postTag, Encoder encoder) {
			super( preTag, postTag,
					"", // don't do any ellipsis to mimic the Elasticsearch behaviour
					false
					// doesn't really matter as we override append() to use encoder rather than rely on this property
			);
			this.encoder = encoder;
		}
```
I haven't added any configuration for the parameter since that would probably be somewhere in the Lucene-specific extension of the DSL, and this PR already has a lot 😄 
By default, Lucene would put `... ` and I've used a "blank" instead.
For the future, we can either expose this as a Lucene-specific config or create our versions of `PassageFormatter` and `FieldHighlighter` to align with the Elasticsearch backend.

Reply:
>  For the future, we can either expose this as a Lucene-specific config or create our versions of PassageFormatter and FieldHighlighter to align with the Elasticsearch backend.

Yes, that seems to make more sense. Or I may misunderstand what this is all about, but still... that deserves some more thought. Can you please create a follow-up issue?

Reply:
https://hibernate.atlassian.net/browse/HSEARCH-4828